README.md

```diff
@@ -53,3 +54,78 @@ usually fine as it is compatible.
+
+You can use the library to check that a user of your domain (e.g. guardian.co.uk) belongs to certain Google Groups. This
+is convenient to create an authorisation system in which only people who are members of some groups can access your
+application.
+
+In order to be able to use the Directory API, you must first set up your own Service account and make sure it has the
+right permissions.
+
+First, log in to your [Google Developer Console](https://console.developers.google.com/), go to your project and create a new service account.
+Follow the instructions given [here](https://developers.google.com/identity/protocols/OAuth2ServiceAccount) to create a service account.
+When asked to download the public/private key pair, choose the JSON option and keep it in a secure place.
+
+You will then need to contact the administrator of your organisation's Google Apps domain and ask them to authorise your service account to access user data in the Google Apps Domain.
+
+They will need to do so by adding specific [scopes](https://developers.google.com/admin-sdk/directory/v1/guides/authorizing) to the service account.
+This is done by granting the *clientId* of the service account with access to the required API scopes.
+
+In order to find out which groups the users of your domain belong to, the only scope you will need is *https://www.googleapis.com/auth/admin.directory.user.readonly*
+
+Finally, you need the email address of a user who has the permission to access the Admin APIs.
+It is not enough to be able to authenticate to your domain with a client ID and private key, you also need to specify the email address of one of the organisation's admin users.
+This email address should be set up to have specific privileges to read groups.
+We call this user the "impersonated user".
+
+As explained in the Google documentation about [Domain-Wide delegation of authority](https://developers.google.com/admin-sdk/directory/v1/guides/delegation):
+
+  > Only users with access to the Admin APIs can access the Admin SDK Directory API, therefore your service account needs to impersonate one of those users to access the Admin SDK Directory API.
+
+Ask the administrator of your organisation's Google Apps domain if you are unsure of what this email address is supposed to be.
+
+Once you have completed those 3 steps, you should be able to integrate it in your application:
+
+- Add the following in your configuration file:
+
+```scala
+serviceaccount.impersonatedUser=adminuser@yourdomain.com
+```
+
+  - Make sure that the service account certificate is accessible
+
+  - This is how you can build your credentials from the json cert file you have downloaded:
+
+```scala
+import java.io.FileInputStream
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential
+import com.gu.googleauth.GoogleServiceAccount
+
+object GoogleAuthConf {
+  val impersonatedUser = current.configuration.getString("serviceaccount.impersonatedUser").get
```

Comment:
You could change this to:

```
val impersonatedUser = ??? // read from config
```

...which lets you drop the `Add the following in your configuration file` section from further up.